cli/azd/pkg/infra/provisioning/manager.go

```diff
@@ -251,6 +305,66 @@ func bindMountOperation(
 	return fileShareService.UploadPath(ctx, subId, shareUrl, source)
 }
 
+func doSqlScriptOperation(
+	ctx context.Context,
+	SqlScriptsOperations []azdOperationSqlServer,
+	console input.Console,
+	env environment.Environment,
+	sqlCmdCli *sqlcmd.SqlCmdCli,
+) error {
+	if len(SqlScriptsOperations) > 0 {
+		console.ShowSpinner(ctx, "execute sql scripts", input.StepFailed)
+	}
+	for _, op := range SqlScriptsOperations {
+		filePath := op.Path
+		if op.Env != nil {
+			fileEnv := environment.NewWithValues("fileEnv", op.Env)
+			tmpDir, err := os.MkdirTemp("", "azd-sql-scripts")
+			if err != nil {
+				return err
+			}
+			defer os.RemoveAll(tmpDir)
+			data, err := os.ReadFile(filePath)
+			if err != nil {
+				return err
+			}
+			expString := osutil.NewExpandableString(string(data))
+			evaluated, err := expString.Envsubst(fileEnv.Getenv)
+			if err != nil {
+				return err
+			}
+			filePath = filepath.Join(tmpDir, filepath.Base(filePath))
+			err = os.WriteFile(filePath, []byte(evaluated), osutil.PermissionDirectory)
+			if err != nil {
+				return err
+			}
+		}
+
+		if _, err := sqlCmdCli.ExecuteScript(
+			ctx,
+			op.Server,
+			op.Database,
+			filePath,
+			// sqlCmd cli uses DAC to connect to the server, but it doesn't know how to handle multi-tenant accounts.
+			// sqlCmd cli asks az or azd for a token w/o passing a tenant-id arg.
+			// sqlCmd cli runs from ~/.azd/bin:
+			//  - azd doesn't know the tenant-id to use and defaults to get a token for home tenant.
+			// By setting the AZURE_SUBSCRIPTION_ID as env var to run sqlCmd cli, azd will use it to get tenant-id.
+			[]string{
+				fmt.Sprintf("%s=%s", environment.SubscriptionIdEnvVarName, env.GetSubscriptionId()),
+			},
+		); err != nil {
+			return fmt.Errorf("error run sqlcmd: %w", err)
+		}
+		console.MessageUxItem(ctx, &ux.DisplayedResource{
+			Type:  sqlServerOperation,
+			Name:  op.Description,
+			State: ux.SucceededState,
+		})
+	}
+	return nil
+}
```

Comment:
Lets chat about this.  There is too much directly coupling here where this component should be for generic provisioning and now handle these special case scenarios.

Reply:
When you say `generic provisioning`, are you reffering to the fact that it doesn't care about how the provisioning implementation is (with bicep, terraform, or other)?

It is not generic in the sense that it can provision to either Azure or another cloud right?

The azd.operations in this model are meant to be an extension of the IaC, for data-plane requests which we want to be executed after creating the resources modeled with IaC.
There is an alternative with Terraform to run local scripts, and bicep is bringing something similar soon. But it is still not that easy to set up authentication and code for the data-plane requests we want.
For example, Terraform allows you to run a local script, but it does not provide you any built-in script for Azure. Something like uploading a file to the storage account you modeled with IaC, or running a query in your DB.  
As a user, you have to bring some azure-sdk, learn how to do what you want and implement the code inside the script.

With azd.operations, I am trying to make it ultra-easy for customers to just use common data-plane operations that we are looking in many templates as hooks with python.